server/api/crud/functions.py

```diff
@@ -153,3 +154,42 @@ def update_function(
             project=project,
             updates=updates,
         )
+
+    def add_function_external_invocation_url(
+        self,
+        db_session: sqlalchemy.orm.Session,
+        function_uri: str,
+        project: str,
+        invocation_url: str,
+    ):
+        _, function_name, tag, hash_key = (
+            mlrun.common.helpers.parse_versioned_object_uri(function_uri)
+        )
+        server.api.utils.singletons.db.get_db().update_function_external_invocation_url(
+            session=db_session,
+            name=function_name,
+            url=invocation_url,
+            project=project,
+            tag=tag,
+            hash_key=hash_key,
+        )
```

Comment:
I know it's the default, but explicitness is worth it here to be resilient to changes
```suggestion
            hash_key=hash_key,
            operation=Operation.ADD,
        )
```